src/Worker.Extensions.DurableTask/DurableTaskClientExtensions.cs

```diff
@@ -143,12 +172,31 @@ static string BuildUrl(string url, params string?[] queryValues)
         //       request headers into consideration and generate the base URL accordingly.
         //       More info: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Forwarded.
         //       One potential workaround is to set ASPNETCORE_FORWARDEDHEADERS_ENABLED to true.
-        string baseUrl = request.Url.GetLeftPart(UriPartial.Authority);
+
+        // If HttpRequestData is provided, use its URL; otherwise, get the baseUrl from the DurableTaskClient.
+        // The base URL could be null if:
+        // 1. The DurableTaskClient isn't a FunctionsDurableTaskClient (which would have the baseUrl from bindings)
+        // 2. There's no valid HttpRequestData provided
+        string? baseUrl = ((request != null) ? request.Url.GetLeftPart(UriPartial.Authority) : GetBaseUrl(client))
+            ?? throw new InvalidOperationException("Base URL is null. Either use Functions bindings or provide an HTTP request to create the HttpPayload.");
+        bool isFromRequest = request != null;
+
         string formattedInstanceId = Uri.EscapeDataString(instanceId);
-        string instanceUrl = $"{baseUrl}/runtime/webhooks/durabletask/instances/{formattedInstanceId}";
+
+        // The baseUrl differs depending on the source. Eg:
+        // - From request: http://localhost:7071/
+        // - From durable client: http://localhost:7071/runtime/webhooks/durabletask
+        // We adjust the instanceUrl construction accordingly.
+        string instanceUrl = isFromRequest
+            ? $"{baseUrl}/runtime/webhooks/durabletask/instances/{formattedInstanceId}"
+            : $"{baseUrl}/instances/{formattedInstanceId}";
         string? commonQueryParameters = GetQueryParams(client);
-        response.Headers.Add("Location", BuildUrl(instanceUrl, commonQueryParameters));
-        response.Headers.Add("Content-Type", "application/json");
+        
+        if (response != null)
+        {
+            response.Headers.Add("Location", BuildUrl(instanceUrl, commonQueryParameters));
+            response.Headers.Add("Content-Type", "application/json");
+        }
```

Comment:
why is `response` possibly `null` now?

Reply:
`SetHeadersAndGetPayload` is also invoked by the existing method [CreateCheckStatusResponse](https://github.com/Azure/azure-functions-durable-extension/blob/dev/src/Worker.Extensions.DurableTask/DurableTaskClientExtensions.cs#L65), which provides an HTTP response to `SetHeadersAndGetPayload`. In the new API, `CreateHttpManagementPayload`, there won’t be an HTTP response parameter. Therefore, when `SetHeadersAndGetPayload` is called by `CreateHttpManagementPayload`, the response will be null; however, if it’s called by `CreateCheckStatusResponse`, the response will not be null.